src/pools/thread/fixed.ts

```diff
@@ -115,7 +115,8 @@ export class FixedThreadPool<
     const listener: (message: MessageValue<Response>) => void = message => {
       if (message.id) {
         const value = this.promiseMap.get(message.id)
-        if (value) {
+        // FIXME this check is really not needed I need to understand why TS ask for it.
+        if (value && value.worker && value.resolve && value.reject) {
```

Comment:
You can simplify it to: `value?.worker && value?.resolve && value?.reject`

Reply:
Please check now I read your comment later but I marked worker, resolve and reject as mandatory on "JustTempValue"

Reply:
> You've deleted too much ...
> The code to benchmark the poolifier pools must stay as well as worker choice algos. Your current work is meant to only replace benchmarking code with other pools:
> 
> * benchmarks/external-pools -> your work on other pools benchmarking
> * benchmarks -> stay as it is, just move the myBench.js and external directory to external-pools.

I did something like we now have two folders with a readme that describe the folders content.
We will go through that readme to add details on benchmarks vs other pools